File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.py
def convert_to_second(time_string):
    splited_time = list(map(int, time_string.split(":")))
    return splited_time[0] * 60 + splited_time[1]

def solution(book_time):
    answer = 0
    time_table = [0 for _ in range(24 * 60 + 10)]
    
    for book in book_time:
        start = book[0]
        end = book[1]
        start_idx = convert_to_second(start)
        end_idx = convert_to_second(end) + 10
        for idx in range(start_idx, end_idx):
            time_table[idx] += 1
    
    return max(time_table)

File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.java
import java.util.*;
import java.lang.Math;

class Solution {
    public int solution(String[][] book_time) {
        int answer = 0;
        int timeTableSize = 24 * 60 + 10;
        int[] timeTable = new int[timeTableSize];
        
        int inputSize = book_time.length;
        
        for(int i = 0; i<inputSize; i++){
            int startIdx = convertToSecond(book_time[i][0]);
            int endIdx = convertToSecond(book_time[i][1]) + 10;
            for (int j = startIdx; j<endIdx; j++){
                timeTable[j]++;
            }
        }
        
        Arrays.sort(timeTable);
        return timeTable[timeTableSize-1];
    }
    
    private int convertToSecond(String time){
        String[] splitedTime = time.split(":");
        return Integer.parseInt(splitedTime[0]) * 60 + Integer.parseInt(splitedTime[1]);
    }
}